src/mlpack/methods/ann/layer/cross_entropy_error_with_logits.hpp

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file cross_entropy_error_with_logits.hpp
+ * @author Kris Singh
+ *
+ * Definition of the cross-entropy with logit performance function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_CROSS_ENTROPY_LOGIT_ERROR_HPP
+#define MLPACK_METHODS_ANN_LAYER_CROSS_ENTROPY_LOGIT_ERROR_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The cross-entropy with logits performance function measures the network's
+ * performance according to the cross-entropy function.
+ * between the input and target distributions.
+ * For more detail look here goo.gl/tRjS6j
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class CrossEntropyErrorLogits
```

Comment:
I think the name `SigmoidCrossEntropyError` suits better than `CrossEntropyErrorLogits`.